helping_functions.py

```diff
@@ -1,171 +1,217 @@
-import pandas as pd
 from osgeo import gdal, ogr
-from osgeo.gdalconst import *
+from osgeo.gdalconst import GA_ReadOnly
+import pandas as pd
+import geopandas as gpd
 import numpy as np
+import shapely
+from shapely import geometry
 import sys
 import datetime
+import inspect
+import os
+
 gdal.PushErrorHandler('CPLQuietErrorHandler')
 
 
 def clean_load_data(paths, param, countries):
-    # Get dataframe with timeseries
-    df_raw = pd.read_excel(paths["load_ts"], header=0, skiprows=[0,1,2], sep=',', decimal='.')
-    # Filter by year
-    df_year = df_raw.loc[df_raw['Year']==param["year"]]
-    
+    """
+
+    :param paths:
+    :param param:
+    :param countries:
+    :return:
+    """
+    timecheck('Start')
+    # avoid reading the excel file each time(to be removed)
+    if not os.path.isfile('savetimeseries_temp.hdf'):
+        # Get dataframe with timeseries
+        print('reading excel file (might take a bit of time)\n')
+        df_raw = pd.read_excel(paths["load_ts"], header=0, skiprows=[0, 1, 2], sep=',', decimal='.')
+        print('done')
+        # Filter by year
+        df_year = df_raw.loc[df_raw['Year'] == param["year"]]
+    else:
+        df_year = pd.read_hdf('savetimeseries_temp.hdf', 'df')
+
     # Scale based on coverage ratio
     df_scaled = df_year.copy()
-    a = df_year.iloc[:,5:].values
-    b = df_year.iloc[:,4].values
-    c = a/b[:, np.newaxis]*100
-    df_scaled.iloc[:,5:] = c
-    del a,b,c
-	
+    a = df_year.iloc[:, 5:].values
+    b = df_year.iloc[:, 4].values
+    c = a / b[:, np.newaxis] * 100
+    df_scaled.iloc[:, 5:] = c
+    del a, b, c
+
     # Reshape so that rows correspond to hours and columns to countries
-    data = np.reshape(df_scaled.iloc[:,5:].values.T,(-1,len(df_scaled['Country'].unique())),order='F')
+    data = np.reshape(df_scaled.iloc[:, 5:].values.T, (-1, len(df_scaled['Country'].unique())), order='F')
     # Create dataframe where rows correspond to hours and columns to countries
     df_reshaped = pd.DataFrame(data, index=np.arange(data.shape[0]), columns=df_scaled['Country'].unique())
-    
-	# Rename countries
+
+    # Rename countries
```

Comment:
I think by renaming them, I solved the problem for UK and EL
Besides, UK is sum of GB and NI